sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/RealtimeNotificationConfigClient.java

```diff
@@ -0,0 +1,123 @@
+            .setHeader("Accept", "application/json");
+
+        // Initialize CountDownLatch and error holder
+        CountDownLatch latch = new CountDownLatch(1);
+        final Throwable[] requestError = { null };
+        final RealTimeNotificationConfig[] configResult = {null};
+
+        // Send the request asynchronously
+        this.httpClient.send(request, CancellationToken.NONE, new HttpCallback() {
+            @Override
+            public void onSuccess(HttpResponse response) {
+                int statusCode = response.getStatusCode();
+                logger.info("Retrieve realtime notification config http response code:" + statusCode);
+                if (statusCode != 200) {
+                    requestError[0] = new RuntimeException("Registrar register request failed with http status code "
+                        + statusCode
+                        + ". Error message: "
+                        + response.getBodyAsString()
+                    );
+                }
+
+                // Convert the response content to RealTimeNotificationConfig
+                try {
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    configResult[0] = objectMapper.readValue(response.getBodyAsString(), RealTimeNotificationConfig.class);
+                    logger.info("Successfully converted response to RealTimeNotificationConfig.");
+                } catch (Exception e) {
+                    logger.error("Failed to parse response body to RealTimeNotificationConfig: " + e.getMessage(), e);
+                    requestError[0] = new RuntimeException("Failed to parse response body", e);
+                }
+
+                latch.countDown();
+            }
+
+            @Override
+            public void onError(Throwable error) {
+                requestError[0] = error;
+                latch.countDown();
+            }
+        });
+
+        awaitOnLatch(latch);
+
+        // Check for errors and throw an exception if necessary
+        if (requestError[0] != null) {
+            throw new RuntimeException(requestError[0]);
+        }
+
+        // Return the result
+        return configResult[0];
+    }
+
+    private boolean awaitOnLatch(CountDownLatch latch) {
+        long timeoutInMin = 1;
+        try {
+            return latch.await(timeoutInMin, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            throw logger.logExceptionAsError(new RuntimeException("Operation didn't complete within " + timeoutInMin + " minutes"));
+        }
+    }
+
+    /**
+     * Converts an InputStream to a String.
+     *
+     * @param is InputStream to convert.
+     * @return String representation of the InputStream.
+     * @throws IOException If an I/O error occurs.
+     */
+    private static String convertStreamToString(InputStream is) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
```

Comment:
Use `try(...) { ... }`

Reply:
btw, are there no common functions that does the same thing?

Reply:
This method was removed in the latest commit. I missed to push the last commit. I changed to use ObjectMapper to convert to the object instead of BufferReader